src/client/java/teammates/client/scripts/DataGenerator.java

```diff
@@ -220,17 +218,15 @@ private static String allInstructors() {
      * Returns Json string presentation for all courses.
      */
     private static String allCourses() {
-        StringBuilder output = new StringBuilder(100);
-        output.append("\"courses\":{\n");
-        for (int i = 0; i < courses.size(); i++) {
-            String course = PREFIX + courses.get(i);
-
-            output.append('\t').append(course(course, "courseIdOf_" + course, "nameOf_" + course));
-            if (i != courses.size() - 1) {
-                output.append(",\n");
-            }
+        StringBuilder outputBuilder = new StringBuilder(100);
+        outputBuilder.append("\"courses\":{\n");
+        for (String course : courses) {
+            String newCourse = PREFIX + course;
+            outputBuilder.append('\t').append(course(newCourse, "courseIdOf_" + newCourse, "nameOf_" + newCourse));
+            outputBuilder.append(",\n");
         }
-        return output.append("\n},").toString();
+        String output = outputBuilder.substring(0, outputBuilder.length() - 2);
+        return output + "\n},";
```

Comment:
Rather than using substring of `length - 2` (2 is a magic number here?), why not just do a regex replace for the end of the string?